### PR TITLE
Add V1 API for getting observation dates based on entity count

### PR DESCRIPTION
**Pull request by shifucun**
Compared with previous version, the major change here is the response shape, where it's oriented by the date with the entity frequency for each facet. This aligns well with its usage, where user wants to know which facet to use for each date.

The dates and count for facets are both sorted.

Also updated the uri and args to match V1 patterns.

### CODE REVIEW

File: internal/server/v1/observationdates/observation_dates_linked.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	store *store.Store,
+) (
+	*pb.BulkObservationDatesLinkedResponse, error) {
+	linkedEntity := in.GetLinkedEntity()
+	entityType := in.GetEntityType()
+	linkedProperty := in.GetLinkedProperty()
+	variables := in.GetVariables()
+	if linkedEntity == "" {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"Missing required argument: ancestor_place")
```
Comment (chejennifer) [inline internal/server/v1/observationdates/observation_dates_linked.go:46]: the argument is linked_entity right?

**Review by chejennifer — APPROVED** [review]
This looks good to me!

File: internal/server/v1/observationdates/observation_dates_linked.go (hunk 1 above)
Comment (shifucun) [reply internal/server/v1/observationdates/observation_dates_linked.go]: Fixed

File: proto/v1/observation_dates.proto [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+option go_package = "./proto";
+package datacommons.v1;
+
+import "stat.proto";
+
+message EntityCount {
```
Comment (pradh) [inline proto/v1/observation_dates.proto:21]: Add some comments to these protos?  e.g., count of what?

**Review by pradh — APPROVED** [review]

File: proto/v1/observation_dates.proto (hunk 2 above)
Comment (shifucun) [reply proto/v1/observation_dates.proto:22]: Added. Thanks!

**Comment by shifucun** [comment]
Thanks for review!

**Pull request merged**